hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/server/StorageContainerManager.java

```diff
@@ -333,7 +332,7 @@ public StorageContainerManager(OzoneConfiguration conf,
         eventQueue);
     blockProtocolServer = new SCMBlockProtocolServer(conf, this);
     clientProtocolServer = new SCMClientProtocolServer(conf, this);
-    httpServer = new StorageContainerManagerHttpServer(conf);
+    httpServer = new StorageContainerManagerHttpServer(conf, this);
```

Comment:
Passing half initialized this to another constructor can cause tricky timing related bugs that are hard to debug at runtime. For example, the metrics are registered on line 368 after line 335. 

Based on the servlet code, we can minimize the dependency with just MetaStore and the DBCheckPointMetrics here, and set/get them into/from the webAppContext. 

    initialize(scm.getScmMetadataStore().getStore(),
        scm.getMetrics().getDBCheckpointMetrics());

Reply:
Hi Xiaoyu, thanks for the comment. I didn't quite get this:

> Based on the servlet code, we can minimize the dependency with just MetaStore and the DBCheckPointMetrics here, and set/get them into/from the webAppContext.

Could you elaborate a bit on this? Thanks!

Reply:
@xiaoyuyao I decided to move `httpServer = new StorageContainerManagerHttpServer(conf, this);` from SCM constructor to `start()` -- the same approach as we already do for OM:

https://github.com/apache/ozone/blob/d964cc98adeaa42d6e6ac34f4fcc591171ee7ff3/hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OzoneManager.java#L1187

This should avoid the race condition. Pls take a look when you are available. :)

I have tested locally that http://127.0.0.1:9876/dbCheckpoint downloads the SCM DB archive for me.